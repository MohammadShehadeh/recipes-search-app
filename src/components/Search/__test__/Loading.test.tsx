import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';

import { Loading } from '..';

describe('Loading', () => {
  it('should render to match snapshot', () => {
    const tree = renderer.create(<Loading />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render Loading correctly', () => {
    render(<Loading length={6} data-testid="loading"  />);

    const LoadingElements = screen.queryAllByTestId("loading");
    expect(LoadingElements.length).toBe(6);
  });
});
