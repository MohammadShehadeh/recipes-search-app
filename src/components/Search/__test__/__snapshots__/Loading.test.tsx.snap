// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading should render to match snapshot 1`] = `
<div
  className="grid"
>
  <div
    className="col col-sm-12 col-md-6 col-lg-4"
  >
    <div
      className="skeleton"
    >
      <div
        className="loading"
        style={Object {}}
      />
    </div>
  </div>
  <div
    className="col col-sm-12 col-md-6 col-lg-4"
  >
    <div
      className="skeleton"
    >
      <div
        className="loading"
        style={Object {}}
      />
    </div>
  </div>
  <div
    className="col col-sm-12 col-md-6 col-lg-4"
  >
    <div
      className="skeleton"
    >
      <div
        className="loading"
        style={Object {}}
      />
    </div>
  </div>
  <div
    className="col col-sm-12 col-md-6 col-lg-4"
  >
    <div
      className="skeleton"
    >
      <div
        className="loading"
        style={Object {}}
      />
    </div>
  </div>
</div>
`;
